esx_service/vmdk_ops.py

```diff
@@ -159,17 +159,74 @@ def RunCommand(cmd):
 
     return (s, o)
 
+
+# Translates the vmdk_path into a readable path by
+# removing the UUID string and using tenant_name
+def gets_readable_path(vmdk_path):
+    idx = 0
+    t_name = ''
+    t_uuid = ''
+    words = vmdk_path.split('/')
+    for word in words:
+        if word == DOCK_VOLS_DIR:
+            break
+        idx += 1
+
+    # The format is not always the same for vmdk_path
+    # If you suspect this is not the known format, return
+    # the original string
+    if (len(words) <= idx + 1):
+        return vmdk_path
+    t_uuid = words[idx + 1]
+    searchObj = re.search(r'[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}',
+                        t_uuid,
+                        re.M|re.I)
+    if searchObj is None:
+        return vmdk_path
+
+    # iterate over the directory and figure out
+    # which symlink's data matches the t_uuid
+    dvpath = ''
+    for i in range(1, idx + 1):
+        dvpath += '/'
+        dvpath += words[i]
+    dvcontents = os.listdir(dvpath)
+    for k in dvcontents:
+        if os.path.islink(dvpath + '/' + k):
+            if t_uuid in os.readlink(dvpath + '/' + k):
+                t_name = k
+
+    if t_name == '':
+        logging.warning("Failed to find tenant name in vmdk_path")
+        return ''
+    dvpath += '/'
+    dvpath += t_name
+    for i in range (idx + 2, len(words)):
+        dvpath += '/'
+        dvpath += words[i]
+    return dvpath
+
+
 # returns error, or None for OK
 # opts is  dictionary of {option: value}.
 # for now we care about size and (maybe) policy
 def createVMDK(vmdk_path, vm_name, vol_name,
                opts={}, vm_uuid=None, tenant_uuid=None, datastore_url=None):
-    logging.info("*** createVMDK: %s opts = %s vm_name=%s tenant_uuid=%s datastore_url=%s",
-                  vmdk_path, opts, vm_name, tenant_uuid, datastore_url)
+
+    e_info, t_uuid, tenant_name = auth.get_tenant(vm_uuid)
```

Comment:
why dd you add get_tenant here ?   It's a fairly expensive op (goes to DB) , so doing it for logging info only is not worth it